File: software/examples/rust/example_simple.rs
use std::{io, error::Error};

use tinkerforge::{ip_connection::IpConnection, 
                  pressure_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Pressure Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let p = PressureBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

		// Get current pressure.
let pressure = p.get_pressure().recv()?;
		println!("Pressure: {} kPa", pressure as f32 /1000.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"tinkerforge/ipconnection"
	"tinkerforge/pressure_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Pressure Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	p, _ := pressure_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current pressure.
	pressure, _ := p.GetPressure()
	fmt.Printf("Pressure: %f kPa\n", float64(pressure)/1000.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
